js/scripts1.js

```diff
@@ -0,0 +1,58 @@
+class General {
+  constructor(name) {
+    this.name = name;
+  }
+
+  getName() {
+    return this.name;
+  }
+}
+
+const drinkCola = {name: 'cola', price: 50, calories: 40};
+const drinkCoffee = {name: 'coffee', price: 80, calories: 20};
+
+class Drink extends General {
+  constructor(name) {
+    super(name);
+  }
+
+  calculatePrice() {
+    return this.getName().price;
+  }
+
+  calculateCalories() {
+    return this.getName().calories;
+  }
+}
+
+const saladCesar = {name: 'cesar', price: 100, calories: 20};
+const saladOliver = {name: 'oliver', price: 50, calories: 80};
+
+class Salad extends General {
+  constructor(name, weight) {
+    super(name);
+    this.weight = weight;
+  }
+
+  getWeight() {
+    return this.weight;
+  }  
+
+  changeWeight(amount) {
+    if(amount != 100) {
```

Comment:
So, you can modify salad, but not `Drink` or `Hamburger`. It looks inconsistent.